neuror/sanitize.py

```diff
@@ -123,3 +129,85 @@ def fix_non_zero_segments(neuron):
     for section in to_be_deleted:
         neuron.delete_section(section)
     return neuron
+
+
+def annotate_neurolucida(morph_path, checkers=None):
+    """Annotate errors on a morphology in neurolucida format.
+
+    Args:
+        morph_path (str): absolute path to an ascii morphology
+        checkers (dict): dict of checker functons from neurom with function as keys
+            and marker data in a dict as values, if None, default checkers are used
+
+    Default checkers include:
+        - fat ends
+        - z-jumps
+        - narrow start
+        - dangling branch
+        - multifurcation
+
+    Returns:
+        annotations to append to .asc file
+        dict of error summary
+        dict of error markers
+    """
+    if checkers is None:
+        checkers = {
+            nc.has_no_fat_ends: {"name": "fat end", "label": "Circle3", "color": "Blue"},
+            partial(nc.has_no_jumps, axis="z"): {
+                "name": "zjump",
+                "label": "Circle2",
+                "color": "Green",
+            },
+            nc.has_no_narrow_start: {"name": "narrow start", "label": "Circle1", "color": "Blue"},
+            nc.has_no_dangling_branch: {"name": "dangling", "label": "Circle6", "color": "Magenta"},
+            nc.has_multifurcation: {
+                "name": "Multifurcation",
+                "label": "Circle8",
+                "color": "Yellow",
+            },
+        }
+
+    def _try(checker, neuron):
+        """Try to apply a checker, and return False if exception raised"""
+        try:
+            return checker(neuron)
+        except Exception as e:  # pylint: disable=broad-except
+            L.exception("%s failed on %s", checker, morph_path)
+            L.exception(e, exc_info=True)
+            return CheckResult(True)
+
+    neuron = load_neuron(morph_path)
+    results = [_try(checker, neuron) for checker in checkers]
+    markers = [
+        dict(setting, data=result.info)
+        for result, setting in zip(results, checkers.values())
+        if not result.status
+    ]
+    summary = {
+        setting["name"]: len(result.info)
+        for result, setting in zip(results, checkers.values())
+        if result.info
+    }
+    return annotate(results, checkers.values()), summary, markers
+
+
+def annotate_neurolucida_all(morph_paths):
+    """Annotate errors on a list of morphologies in neurolicida format.
+
+    Args:
+        morph_paths (list): list of str of paths to morphologies.
+
+    Returns:
+        dict annotations to append to .asc file (morph_path as keys)
+        dict of dict of error summary (morph_path as keys)
+        dict of dict of markers (morph_path as keys)
+    """
+    summaries, annotations, markers = {}, {}, {}
+    for morph_path in morph_paths:
```

Comment:
Can you add parallelization the same way other *_all methods in this file are doing ?

Reply:
sure!